### PR TITLE
Simplify TestReconcilerFinalizer cleanup

### PR DESCRIPTION
**Pull request by karlkfi**
- Combine cleanup functions and use lists and for loops
- Skip cleanup on failure if debug is enabled
- Delete orphened safety namespaces

### CODE REVIEW

File: e2e/testcases/reconciler_finalizer_test.go [hunk 1]
```diff
@@ -216,14 +236,22 @@ func TestReconcilerFinalizer_MultiLevelForeground(t *testing.T) {
...
+				fake.RootSyncObjectV1Beta1(rootSyncNN.Name),
+				fake.RepoSyncObjectV1Beta1(repoSyncNN.Namespace, repoSyncNN.Name),
+			},
+			[]client.Object{
+				fake.DeploymentObject(core.Name(deployment1NN.Name), core.Namespace(deployment1NN.Namespace)),
+				fake.DeploymentObject(core.Name(deployment2NN.Name), core.Namespace(deployment2NN.Namespace)),
+				fake.NamespaceObject(namespace1NN.Name),
+				fake.NamespaceObject(safetyNamespace1NN.Name),
+				fake.NamespaceObject(safetyNamespace2NN.Name),
+			})
```
Comment (haiyanmeng) [inline e2e/testcases/reconciler_finalizer_test.go:254]: These lines are also in TestReconcilerFinalizer_MultiLevelMixed. Put them in a helper func.

File: e2e/testcases/reconciler_finalizer_test.go [hunk 2]
```diff
@@ -53,9 +55,18 @@ func TestReconcilerFinalizer_Orphan(t *testing.T) {
...
-		cleanupSingleLevel(nt, rootSyncNN, deployment1NN, namespace1NN)
+		cleanupSyncsAndObjects(nt,
+			[]client.Object{
+				fake.RootSyncObjectV1Beta1(rootSyncNN.Name),
+			},
+			[]client.Object{
+				fake.DeploymentObject(core.Name(deployment1NN.Name), core.Namespace(deployment1NN.Namespace)),
+				fake.NamespaceObject(namespace1NN.Name),
+				fake.NamespaceObject(safetyNamespace1NN.Name),
+			})
```
Comment (haiyanmeng) [inline e2e/testcases/reconciler_finalizer_test.go:69]: These lines are in TestReconcilerFinalizer_Foreground. Put them in a helper func.

File: e2e/testcases/reconciler_finalizer_test.go (hunk 1 above)
Comment (karlkfi) [reply e2e/testcases/reconciler_finalizer_test.go]: done

File: e2e/testcases/reconciler_finalizer_test.go (hunk 2 above)
Comment (karlkfi) [reply e2e/testcases/reconciler_finalizer_test.go]: done

**Review by haiyanmeng — APPROVED** [review]

**Pull request merged**